PSKoans/Koans/Cmdlets 1/AboutPSProviders.Koans.ps1

```diff
@@ -127,17 +127,33 @@ Describe 'Environment Provider' {
 }
 
 Describe 'FileSystem Provider' {
-    $Path = 'TestDrive:' | Join-Path -ChildPath 'File001.tmp'
+    if( $PSVersionTable.PSVersion.Major -le 5 ){
+        #If on windows 5 or earlier, use env temp directory.
+        $Path = Join-Path -Path $env:TEMP -ChildPath 'PSKoans.tmp'
+    }
+    elseif( (Get-Variable -Name IsWindows)){
+        if( $IsWindows){
+            $Path = Join-Path -Path $env:TEMP -ChildPath 'PSKoans.tmp'
+        }
+        elseif( (Test-Path \tmp)){
+            #We found a \tmp directory, let's use it!
+            $Path = Join-Path -Path \tmp -ChildPath 'PSKoans.tmp'
+        }
+        else{
+            #Path to MacOS temp directory
+            $Path = Join-Path -Path $env:TMPDIR -ChildPath 'PSKoans.tmp'
+        }
+    }
+    else{
+        Write-Error -Message "Was not able to determine temp directory location."
+    }
     if (-not (Test-Path $Path)) {
-        New-Item -Path $Path > $null
-
-        -join (1..10000 | ForEach-Object {
-                Get-Random -Minimum 32 -Maximum 96 |
-                    ForEach-Object {[char]$_ -as [string]}
-                if ((Get-Random) -match '[25]0$') {
-                    "`n"
-                }
-            }) | Set-Content -Path $Path
+
+        "PSKOANS!
+        The Env: drive contains system environment data. Its contents can vary wildly from OS to OS,
+        especially between Windows, Mac, and Linux, for example.
+
+        The only shared Env: items across all OS's currently are Path and PSModulePath." | Add-Content -Path $Path
```

Comment:
This might perhaps be better formatted as a here string, I think?

Reply:
I will change this to a here string. I think it is important to keep this snippet as part of the koan, so when reading through the test it is clear what content is in the file. Otherwise, one would have to browse to their temp directory and open a file. Doesn't seem helpful when trying to complete:
https://github.com/vexx32/PSKoans/blob/f18528243e64ccc4006844ca2088e2ee8f7a3e62/PSKoans/Koans/Cmdlets%201/AboutPSProviders.Koans.ps1#L167-L169

Reply:
Agreed.